tree/tree/src/TTree.cxx

```diff
@@ -6122,30 +6122,52 @@ const char* TTree::GetFriendAlias(TTree* tree) const
    if (kGetFriendAlias & fFriendLockStatus) {
       return nullptr;
    }
-   if (!fFriends) {
+   TFriendLock lock(const_cast<TTree *>(this), kGetFriendAlias);
+
+   auto lookForFriendNameInListOfFriends = [tree](const TList &friends) -> const char * {
+      for (auto *frEl : ROOT::Detail::TRangeStaticCast<TFriendElement>(friends)) {
+         auto *frElTree = frEl->GetTree();
+         // Simplest case: we found a friend which tree is the same as the input tree
+         if (frElTree == tree)
+            return frEl->GetName();
+         // Try again: the friend tree might be actually a TChain
+         if (frElTree && frElTree->GetTree() == tree)
+            return frEl->GetName();
+      }
       return nullptr;
+   };
+
+   // First, look for the immediate friends of this tree
+   if (fFriends) {
+      const char *friendAlias = lookForFriendNameInListOfFriends(*fFriends);
+      if (friendAlias)
+         return friendAlias;
    }
-   TFriendLock lock(const_cast<TTree*>(this), kGetFriendAlias);
-   TIter nextf(fFriends);
-   TFriendElement* fe = nullptr;
-   while ((fe = (TFriendElement*) nextf())) {
-      TTree* t = fe->GetTree();
-      if (t == tree) {
-         return fe->GetName();
-      }
-      // Case of a chain:
-      if (t && t->GetTree() == tree) {
-         return fe->GetName();
+
+   // Then, check if this is a TChain and the current tree has friends
```

Comment:
We might  not be able to avoid it but in the usual case, this should be redundant.

If this is a chain and it has friends, it will make its underlying tree have the friends too (so same semantic friends).

The case that is non-redundantly covered here is the case where the underlying TTree of a TChain has its own persistent friends.

If this analysis is correct, we ought to clarify it in the code comment (for example to avoid a future maintainer thinking only of the former case and wanting to remove this code)

Reply:
I believe this is already non-redundant because of the condition
```
innerListOfFriends && innerListOfFriends != fFriends
```
but I'm also adding a comment to make this explicit